substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/Agent.java

```diff
@@ -88,37 +72,34 @@
 import com.oracle.svm.configure.config.ConfigurationSet;
 import com.oracle.svm.configure.filters.FilterConfigurationParser;
 import com.oracle.svm.configure.filters.RuleNode;
+import com.oracle.svm.configure.json.JsonPrintable;
 import com.oracle.svm.configure.json.JsonWriter;
 import com.oracle.svm.configure.trace.AccessAdvisor;
 import com.oracle.svm.configure.trace.TraceProcessor;
 import com.oracle.svm.core.FallbackExecutor;
 import com.oracle.svm.core.SubstrateUtil;
-import com.oracle.svm.core.c.function.CEntryPointOptions;
-import com.oracle.svm.core.c.function.CEntryPointSetup;
 import com.oracle.svm.core.configure.ConfigurationFiles;
 import com.oracle.svm.driver.NativeImage;
 import com.oracle.svm.jni.nativeapi.JNIEnvironment;
-import com.oracle.svm.jni.nativeapi.JNIErrors;
 import com.oracle.svm.jni.nativeapi.JNIJavaVM;
 import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
-import com.oracle.svm.jni.nativeapi.JNIVersion;
 
-public final class Agent {
+public final class Agent extends AgentBase {
```

Comment:
Perhaps rename this to `NativeImageAgent` too. Then you could also rename `AgentBase` to `Agent` if you'd like.

Reply:
Yes, that would be perfect. Changing it.